### ISSUE TITLE
Use of dynamic properties in instrument "channel"

### ISSUE THREAD

**Issue by msmttchr** [pymeasure/pymeasure#616]
Just discovered that dynamic properties are not supported in some implementation of instrument "channel". For example, if we consider the following class: https://github.com/pymeasure/pymeasure/blob/d7547d6cf79b85d8bee96958067537095c116df9/pymeasure/instruments/newport/esp300.py#L130
The properties declared inside the class can be defined dynamic by adding the appropriate parameter, but they will not behave as dynamic properties and the original value cannot be overridden.
I think this needs to be taken into account when some guidelines for Channels implementation are provided (see https://github.com/pymeasure/pymeasure/issues/416#issuecomment-1025090693)

**Comment by bilderbuchi** [comment]
Good catch! This could be because it's missing something that is coming from `Instrument`, which we normally inherit from. Maybe we should see if we can extract the common/needed bits into a superclass for `Instrument`, and implement the `BaseChannel` that comment mentions, which then also inherits from that class to get this all working as intended?

**Comment by bilderbuchi** [comment]
Does it fail "visibly" or does it just not work as expected?

**Comment by msmttchr** [comment]
> Good catch! This could be because it's missing something that is coming from `Instrument`, which we normally inherit from. Maybe we should see if we can extract the common/needed bits into a superclass for `Instrument`, and implement the `BaseChannel` that comment mentions, which then also inherits from that class to get this all working as intended?

I agree, think this could be a good approach.

**Comment by msmttchr** [comment]
> Does it fail "visibly" or does it just not work as expected?

It just doesn't work, but there is no visible failure.
I was logging it, just to take into account this aspect for the restructuring of `Instrument` class.

**Comment by msmttchr** [comment]
See also #649 for a possible solution to this problem

**Comment by bilderbuchi** [comment]
> See also #649 for a possible solution to this problem

I have not reviewed this in detail, so I might not know what you refer to exactly, but I think #649 shows a workaround (that will probably bite us later), not a solution (see https://github.com/pymeasure/pymeasure/pull/649#issuecomment-1202467447).

I think implementing a `BaseChannel`/`ChannelBase` as outlined above could resolve this, too, so we should make sure not to forget to

* [ ] fix the AWG-401x workaround for dynamic properties in channels.

Now that #634 is merged (finally! :sweat_smile: ), it should be easier to put some protocol tests in place to nail down the exchanged messages, including dynamicness etc, and then refactor all those classes inheriting from `Instrument`, without having the device present. :-)
